File: Assets/ExitButton.js
#pragma strict

var gameManager : GameManager;
var level : String;
var sndSelect : AudioClip;


function OnMouseDown() {
    var ray : Ray = Camera.main.ScreenPointToRay(Input.mousePosition);
    var hit : RaycastHit2D = Physics2D.GetRayIntersection(ray,Mathf.Infinity);
    var wasReallyHit : boolean = false;  
    if ((hit.collider != null) && (hit.collider.transform == transform))
    {
    	wasReallyHit = true;
    }
    // fix bug where onmousedown is triggered when it shouldn't be
	if (!wasReallyHit) return;
	
	audio.PlayOneShot(sndSelect,1.0);
	if (gameManager != null) {
		gameManager.Cleanup();
	}
	Application.LoadLevel(level);
	Debug.Log("load level " + level);
	Destroy(this.gameObject);
}

function OnMouseUp() {
}

function Start () {

}

function Update () {

}

function OnDrawGizmos() {
	var b = collider2D.bounds;
	var t = transform;
	
	// Draw BoxColliders
    if (b != null) { 
		var tl = new Vector3(t.position.x - (b.size.x / 2), t.position.y + (b.size.y / 2), 0f);
        var bl = new Vector3(t.position.x - (b.size.x / 2), t.position.y - (b.size.y / 2), 0f);
        var br = new Vector3(t.position.x + (b.size.x / 2), t.position.y - (b.size.y / 2), 0f);
        var tr = new Vector3(t.position.x + (b.size.x / 2), t.position.y + (b.size.y / 2), 0f);
        Gizmos.color = Color.red;
        Gizmos.DrawLine (tl, bl);
        Gizmos.DrawLine (bl, br);
        Gizmos.DrawLine (br, tr);
        Gizmos.DrawLine (tr, tl);
    }
}

File: Assets/cs/ExitButton.cs
namespace CSDEP
{
using UnityEngine;
using System;
using UnityEngine.SceneManagement;


public class ExitButton:MonoBehaviour{
	
	public GameManager gameManager;
	public string level;
	public AudioClip sndSelect;
	
	
	public void OnMouseDown() {
	    Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
	    RaycastHit2D hit = Physics2D.GetRayIntersection(ray,Mathf.Infinity);
	    bool wasReallyHit = false;  
	    if ((hit.collider != null) && (hit.collider.transform == transform))
	    {
	    	wasReallyHit = true;
	    }
	    // fix bug where onmousedown is triggered when it shouldn't be
		if (!wasReallyHit)  {
			Debug.Log("exit button not really hit");
			return;
		}
		
		if (gameManager != null) {
			gameManager.Cleanup();
		}
//		Application.LoadLevel(level);
		SceneManager.LoadScene(level);
		Debug.Log("load level " + level);
	}
	
	public void OnMouseUp() {
	}
	
	public void Start() {
		SpriteRenderer sr = GetComponent<SpriteRenderer>();
		sr.color = new Color(80.0f/255.0f, 89.0f/255.0f, 184.0f/255.0f, 1f);
	}
	
	public void Update() {
	
	}

}
}
